Sources/AudioKit/Nodes/Playback/AudioPlayer/AudioPlayer+Playback.swift

```diff
@@ -34,36 +23,47 @@ extension AudioPlayer {
             Log("🛑 Error: AudioPlayer's engine must be running before playback.", type: .error)
             return
         }
+        switch status {
+        case .stopped:
+            editStartTime = startTime ?? editStartTime
+            editEndTime = endTime ?? editEndTime
 
-        if when != nil {
-            scheduleTime = nil
-            playerNode.stop()
-        }
-
-        editStartTime = startTime ?? editStartTime
-        editEndTime = endTime ?? editEndTime
-
-        if !isScheduled || isSeeking {
             schedule(at: when,
                      completionCallbackType: completionCallbackType)
-        }
 
-        playerNode.play()
-        isPlaying = true
+            playerNode.play()
+            status = .playing
+        case .playing:
+            // player is already playing
+            return
+        case .paused:
+            resume()
+        case .scheduling:
+            // player is already scheduling
+            return
+        case .completed:
```

Comment:
This replaces the logic inside the completion handler -- just moved most of it here.